apps/api/src/persistence/migrations/01/remove-base-digital-asset-url.migration.spec.ts

```diff
@@ -0,0 +1,164 @@
+import createTestModule from '../../../app/controllers/__tests__/createTestModule';
+import { Photograph } from '../../../domain/models/photograph/entities/photograph.entity';
+import { Term } from '../../../domain/models/term/entities/term.entity';
+import buildDummyUuid from '../../../domain/models/__tests__/utilities/buildDummyUuid';
+import { ResourceType } from '../../../domain/types/ResourceType';
+import { InternalError } from '../../../lib/errors/InternalError';
+import { DTO } from '../../../types/DTO';
+import { ArangoConnectionProvider } from '../../database/arango-connection.provider';
+import { ArangoQueryRunner } from '../../database/arango-query-runner';
+import { ArangoCollectionId } from '../../database/collection-references/ArangoCollectionId';
+import { ArangoDatabaseProvider } from '../../database/database.provider';
+import { DatabaseDocument, DatabaseDTO } from '../../database/utilities/mapEntityDTOToDatabaseDTO';
+import generateDatabaseNameForTestSuite from '../../repositories/__tests__/generateDatabaseNameForTestSuite';
+import TestRepositoryProvider from '../../repositories/__tests__/TestRepositoryProvider';
+import { RemoveBaseDigitalAssetUrl } from './remove-base-digital-asset-url.migration';
+
+describe(`RemoveBaseDigitalAssetUrl`, () => {
+    let testDatabaseProvider: ArangoDatabaseProvider;
+
+    let testQueryRunner: ArangoQueryRunner;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    beforeAll(async () => {
+        const testModule = await createTestModule({
+            ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+        });
+
+        const arangoConnectionProvider = testModule.get(ArangoConnectionProvider);
+
+        testDatabaseProvider = new ArangoDatabaseProvider(arangoConnectionProvider);
+
+        /**
+         * It's a bit awkward that we need this because we are not working at
+         * the repositories level of abstraction. However, we have added test
+         * setup and teardown logic at this level for the purpose of commadn and
+         * query integration tests. So instead of rewriting this logic on a
+         * `TestDatabaseProvider`, we will just leverage this existing logic for
+         * test teardown.
+         */
+        testRepositoryProvider = new TestRepositoryProvider(testDatabaseProvider);
+
+        testQueryRunner = new ArangoQueryRunner(testDatabaseProvider);
+    });
+
+    const baseDigitalAssetUrl = `https://www.mymedia.org/downloads/`;
+
+    const buildId = (resourceType: ResourceType, index: number): string => {
+        if (resourceType === ResourceType.term) return buildDummyUuid(index);
+
+        if (resourceType === ResourceType.photograph) return buildDummyUuid(100 + index);
+
+        throw new InternalError(
```

Comment:
Is this temporary as you work through types?

Reply:
I suppose that's another way of asking whether this could be broken out as a reusable test utility. I can definitely see that, but I'll wait til we hit the need. In that case, I may want to alphabetize and use a sequence number to get a unique ID, except that that will cause the IDs to change so they won't be good if we are snapshotting...

Reply:
Maybe a more robust way is to curry the helper, providing a map from resource type to offset and fallback to the logic I mentioned above as default.